### PR TITLE
fv3fit: wandb integration

### PR DESCRIPTION
**Pull request by AnnaKwa**
This adds an option to log `fv3fit.train` runs using wandb. The project, entity, and job type are automatically initialized using the  environment variables `WANDB_ENTITY, WANDB_PROJECT, WANDB_JOB_TYPE`.

Right now the only training that has logging setup is the dense model training, which will log the end of epoch validation loss. In the future, the wandb keras callback can be used; however, the training loop code needs some refactoring before it can use arbitrary keras callbacks.

Tested this out with my own local environment:  https://wandb.ai/annakwa/test-fv3fit/runs/27mfwt81

Added public API:
- `--wandb` flag for `fv3fit.train`

### CODE REVIEW

**Review by nbren12 — CHANGES_REQUESTED** [review]
super excited to see this capability, and happy that is has integrated so well.

In principle, this should probably be a 1-2 line change at the highest level of the code e.g.
```

# not 100% sure this will work
wandb.config(config=config_dict)

# fv3fit.train should be modified to enable this 1-liner and allow adding callbacks more dynamically
if wandb.initialized:
    callbacks.append(wandb.keras.Callback())
    
#... somewhere else
model.fit(..., callbacks=callbacks)  
```

File: external/fv3fit/fv3fit/keras/_models/shared/callbacks.py
Comment (AnnaKwa) [reply external/fv3fit/fv3fit/keras/_models/shared/callbacks.py:45]: > .fit_loop should be modified to take a list of tf.keras.callbacks.Callback objects if it doesn't already.

Yes, that was the issue with using the wandb keras callback in this PR- currently `.fit_loop` does not take a list of keras callbacks but instead wants a list of `Callable[[EpochResult], None]` which is a remnant of how it used to call `.fit` every epoch. I talked to Jeremy yesterday and decided that I was going to change `fit_loop` to just take in keras callbacks and remove the `EpochCallback` class. It seems out of the scope of this PR though. In that future PR, this class will be removed. Is it ok to leave this for now since it will be removed but will allow me to run sweeps in the meantime?

Comment (AnnaKwa) [reply external/fv3fit/fv3fit/keras/_models/shared/callbacks.py:45]: Or, I could put this PR on hold until the one remove `EpochCallback` is merged.

Comment (nbren12) [reply external/fv3fit/fv3fit/keras/_models/shared/callbacks.py:45]: No problem if you are planning on submitting another pr soon. it's not hard to undo these changes.

File: external/fv3fit/fv3fit/train.py
Comment (nbren12) [reply external/fv3fit/fv3fit/train.py:200]: If this doesn't work then feel free to keep the old code...I'm changing to approve.

Comment (AnnaKwa) [reply external/fv3fit/fv3fit/train.py:200]: Yes, I tried the suggested version, the sweep doesn't support `.` to access keys in nested dicts. Apparently there were some requests and it was a planned feature at some point (https://github.com/wandb/client/issues/982) but I don't think they've added it yet.

**Pull request merged**
